pkg/client/connector/service.go

```diff
@@ -504,9 +505,16 @@ func run(c context.Context) error {
 		if !ok {
 			return nil
 		}
-		tm.sshPortForward(c,
-			"-D", "localhost:1080",
-		)
+		ln, err := net.Listen("tcp4", "127.0.0.1:0")
+		if err != nil {
+			return err
+		}
+		dynamicAddr := ln.Addr().String()
+		_ = ln.Close()
+		_, portStr, _ := net.SplitHostPort(dynamicAddr)
+		dynamicPort, _ := strconv.Atoi(portStr)
+		tm.dynamicSshPort = int32(dynamicPort)
+		tm.sshPortForward(c, "-D", dynamicAddr)
```

Comment:
Any reason not to use `-D localhost:0` and parse the output like we do with the `kubectl port-forward`s?

Reply:
That behaviour is only for `-R`. Trying it with `-D` results in `"Bad dynamic forwarding specification 'localhost:0'"` (which is consistent with the [man page](https://man.openbsd.org/ssh).